File: RepoTransBench/target_projects/Python/C++/stephenmcd_django-forms-builder/public_tests/test_public_fields.cpp
#include <gtest/gtest.h>
#include "forms_fields.h"

using namespace forms_fields;

TEST(PublicField, ToPythonBooleanTrue) {
    BooleanField boolean_field;
    EXPECT_TRUE(boolean_field.to_python("on"));
}

TEST(PublicField, ToPythonBooleanFalse) {
    BooleanField boolean_field;
    EXPECT_FALSE(boolean_field.to_python(""));
    EXPECT_FALSE(boolean_field.to_python(nullptr));
}

TEST(PublicField, ToPythonSelect) {
    SelectField select_field("orange|banana|pear");
    std::vector<std::string> expected{"orange", "banana", "pear"};
    EXPECT_EQ(select_field.choices_, expected);
}

TEST(PublicField, PrettyNameWithNumber) {
    EXPECT_EQ(pretty_name("item_123_value"), "Item 123 value");
}

File: RepoTransBench/target_projects/Python/Rust/stephenmcd_django-forms-builder/public_tests/test_public_fields.rs
use crate::forms_builder::forms::fields;

#[test]
fn test_public_field_to_python_boolean_true() {
    let boolean_field = fields::BooleanField;
    assert!(boolean_field.to_python(Some("on")));
}

#[test]
fn test_public_field_to_python_boolean_false() {
    let boolean_field = fields::BooleanField;
    assert!(!boolean_field.to_python(Some("")));
    assert!(!boolean_field.to_python(None));
}

#[test]
fn test_public_field_to_python_select() {
    let select_field = fields::SelectField::new("orange|banana|pear");
    assert_eq!(select_field.choices, vec!["orange", "banana", "pear"]);
}

#[test]
fn test_public_pretty_name_with_number() {
    let result = fields::pretty_name("item_123_value");
    assert_eq!(result, "Item 123 value");
}